tests/stores/test_azure.py

```diff
@@ -0,0 +1,434 @@
+"""
+Azure testing requires Azurite.
+It can be set up according to the instructions:
+https://github.com/Azure/Azurite
+With docker can be started by running:
+docker run -p 10000:10000 mcr.microsoft.com/azure-storage/azurite azurite-blob --blobHost 0.0.0.0
+"""
+
+import time
+from datetime import datetime
+from contextlib import contextmanager
+import pytest
+
+from maggma.stores import MemoryStore, MongoStore, AzureBlobStore
+
+try:
+    import azure.storage.blob as azure_blob
+    from azure.storage.blob import BlobServiceClient
+except (ImportError, ModuleNotFoundError):
+    azure_blob = None  # type: ignore
+
+
+AZURITE_CONNECTION_STRING = (
+    "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;"
+    "AccountKey="
+    "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq"
+    "/K1SZFPTOtr/KBHBeksoGMGw==;"
+    "BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;"
+)
+
+AZURITE_CONTAINER_NAME = "maggma-test-container"
+
+
+@pytest.fixture
+def mongostore():
+    store = MongoStore("maggma_test", "test")
+    store.connect()
+    yield store
+    store._collection.drop()
+
+
+# a context manager and not a fixture to handle multiple containers in the
```

Comment:
Is it likely to have multiple containers in the same test function ? If not, then we can use pytest.fixture(scope="function") that I think reinstantiate everytime. Anyway, it is not super important.

Reply:
There is indeed a test where two different containers are used